server/src/utils/media.ts

```diff
@@ -1014,14 +1023,18 @@ export class RkmppHwDecodeConfig extends RkmppSwDecodeConfig {
 
   getFilterOptions(videoStream: VideoStreamInfo) {
     if (this.shouldToneMap(videoStream)) {
-      const colors = this.getColors();
-      return [
-        `scale_rkrga=${this.getScaling(videoStream)}:format=p010:afbc=1`,
-        'hwmap=derive_device=opencl:mode=read',
-        `tonemap_opencl=format=nv12:r=pc:p=${colors.primaries}:t=${colors.transfer}:m=${colors.matrix}:tonemap=${this.config.tonemap}:desat=0`,
-        'hwmap=derive_device=rkmpp:mode=write:reverse=1',
-        'format=drm_prime',
-      ];
+      if (this.hasMaliOpenCL) {
+        const colors = this.getColors();
+        return [
+          `scale_rkrga=${this.getScaling(videoStream)}:format=p010:afbc=1`,
+          'hwmap=derive_device=opencl:mode=read',
+          `tonemap_opencl=format=nv12:r=pc:p=${colors.primaries}:t=${colors.transfer}:m=${colors.matrix}:tonemap=${this.config.tonemap}:desat=0`,
+          'hwmap=derive_device=rkmpp:mode=write:reverse=1',
+          'format=drm_prime',
+        ];
+      } else {
+        return super.getFilterOptions(videoStream);
+      }
```

Comment:
```suggestion
      }
      return super.getFilterOptions(videoStream);
```
Nit: you don't need `else` here since the if block returns.